airbyte-cdk/python/unit_tests/sources/test_connector_state_manager.py

```diff
@@ -0,0 +1,34 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+import pytest
+from airbyte_cdk.models import AirbyteStateMessage, AirbyteStateType
+from airbyte_cdk.sources.connector_state_manager import ConnectorStateManager
+
+
+@pytest.mark.parametrize(
+    "test_name, input_state, expected_legacy_state",
+    [
+        (
+            "test_legacy_input_state",
+            [AirbyteStateMessage(type=AirbyteStateType.LEGACY, data={"actresses": {"id": "seehorn_rhea"}})],
+            {"actresses": {"id": "seehorn_rhea"}},
+        ),
+        (
+            "test_supports_legacy_json_blob",
+            {
+                "actors": {"created_at": "1962-10-22"},
+                "actresses": {"id": "seehorn_rhea"},
+            },
+            {"actors": {"created_at": "1962-10-22"}, "actresses": {"id": "seehorn_rhea"}},
+        ),
+        ("test_initialize_empty_mapping_by_default", {}, {}),
+        ("test_initialize_empty_state", [], {}),
+    ],
```

Comment:
I'm not sure the first element of the tuple is used as a test id for Pytest
```suggestion
    " input_state, expected_legacy_state",
    [
        pytest.params(
            [AirbyteStateMessage(type=AirbyteStateType.LEGACY, data={"actresses": {"id": "seehorn_rhea"}})],
            {"actresses": {"id": "seehorn_rhea"}},
            id="test_legacy_input_state"
        ),
        ...
    ],
```